airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -0,0 +1,677 @@
+        # add parent_stream_state to `updated_state`
+        updated_state[self.parent_stream.name] = stream_state_cache.cached_state.get(self.parent_stream.name)
+        return updated_state
+
+    def request_params(self, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = {"limit": self.limit}
+        if next_page_token:
+            params.update(**next_page_token)
+        return params
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        """
+        Reading the parent stream for slices with structure:
+        EXAMPLE: for given nested_record as `id` of Orders,
+
+        Outputs:
+            [
+                {slice_key: 123},
+                {slice_key: 456},
+                {...},
+                {slice_key: 999
+            ]
+        """
+        sorted_substream_slices = []
+
+        # reading parent nested stream_state from child stream state
+        parent_stream_state = stream_state.get(self.parent_stream.name) if stream_state else {}
+
+        # reading the parent stream
+        for record in self.parent_stream.read_records(stream_state=parent_stream_state, **kwargs):
+            # updating the `stream_state` with the state of it's parent stream
+            # to have the child stream sync independently from the parent stream
+            stream_state_cache.cached_state[self.parent_stream.name] = self.parent_stream.get_updated_state({}, record)
+            # to limit the number of API Calls and reduce the time of data fetch,
+            # we can pull the ready data for child_substream, if nested data is present,
+            # and corresponds to the data of child_substream we need.
+            if self.nested_substream and self.nested_substream_list_field_id:
+                if record.get(self.nested_substream):
+                    sorted_substream_slices.extend(
+                        [
+                            {
+                                self.slice_key: sub_record[self.nested_substream_list_field_id],
+                                self.cursor_field: record[self.nested_substream][0].get(
+                                    self.cursor_field, self.default_state_comparison_value
+                                ),
+                            }
+                            for sub_record in record[self.nested_record]
+                        ]
+                    )
+            elif self.nested_substream:
+                if record.get(self.nested_substream):
+                    sorted_substream_slices.append(
+                        {
+                            self.slice_key: record[self.nested_record],
+                            self.cursor_field: record[self.nested_substream][0].get(self.cursor_field, self.default_state_comparison_value),
+                        }
+                    )
+            else:
```

Comment:
what's the use case for using a nested substream class without a None nested_substream? should we use a non-nested IncrementalShopifySubstream?